### DISCUSSION TITLE
How not to expose the code when we upgrade the contract

### DISCUSSION THREAD

**Discussion by neethu-satheesh-s** [aptos-labs/aptos-developer-discussions#287] [Questions]
### Discord user ID

neethu.satheesh

### Describe your question in detail.

Hi,
I have a contract which was published via resource account and not to expose the code. The contract is an upgradable contract.

`aptos move create-resource-account-and-publish-package --seed 1234 --address-name abc --named-addresses source_addr=12347d47a9b0ac564856168b68f5f06408cc5f1c691yur5366c3ab116d76rsdf --included-artifacts none --profile admin`

included-artifacts none ensures the code does not get exposed.

But when I upgrade using the following command, it gets exposed. 
`aptos move build-publish-payload --json-output-file upgrade.json --named-addresses source_addr=12347d47a9b0ac564856168b68f5f06408cc5f1c691yur5366c3ab116d76rsdf`

Then edited the function to match the update function defined in the contract.

How do I call the upgrade function without exposing the code? 
Thank you.

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

N/A

**Comment by adityaaggarwal12** [comment]
you should be able to include the `--included-artifacts none` option in the `aptos move build-publish-payload` command. Like this

```sh
aptos move build-publish-payload 
--json-output-file upgrade.json 
--named-addresses source_addr=12347d47a9b0ac564856168b68f5f06408cc5f1c691yur5366c3ab116d76rsdf 
--included-artifacts none
```
This way, you won’t expose your code on-chain during the upgrade.

**Reply by harispericorn** [reply]
--included-artifacts none is not a option in build-publish-payload

**Reply by gregnazario** [reply]
It should be, what's your CLI version?  You may need to upgrade it https://aptos.dev/en/build/cli
